### PR TITLE
Complete overarching UML diagram and earmark key classes for deletion

### PR DESCRIPTION
**Pull request by sharadhr**
UML diagram has been added. Do feel free to comment.

### CODE REVIEW

File: docs/diagrams/BasicRecipeDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,125 @@
+@startuml BasicRecipeDiagram
+show members
+
+hide circle
+skinparam classAttributeIconSize 0
+
+class RecipeBook {
+    -recipes : UniqueRecipeList
+    +list()
+    +duplicate()
```
Comment (kevinswk94) [inline docs/diagrams/BasicRecipeDiagram.puml:12]: what about methods to add / delete Recipes to the Recipe list?
+addRecipe()
+deleteRecipe()

**Review by kevinswk94 — APPROVED** [review]
Looks fine to me! other than the 1 small suggestion.

File: docs/diagrams/BasicRecipeDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    -name : String
+    -symbol : UnitSymbol
+    {static} +convertTo()
+}
+
+enum UnitSymbol <<enumeration>> {
+    tsp
+    tbp
+    cup
+    ml
```
Comment (AdarshChugani) [inline docs/diagrams/BasicRecipeDiagram.puml:83]: Not sure if enum is the best idea for units, cause there are a lot.

If you do want to go with enum, I think these should be added too

* Pint
* Gallon
* Quarts
* Litres
* Stick (for stuff like butter)

**Review by AdarshChugani — APPROVED** [review]
Apart from what Kevin said, you may want to reconsider using the enum for units.

Other than that, it looks good!

File: docs/diagrams/BasicRecipeDiagram.puml (hunk 2 above)
Comment (sharadhr) [reply docs/diagrams/BasicRecipeDiagram.puml:83]: How else could we collect all possible units? Noted on the pint, gallon, quarts, litres, sticks, etc.

Comment (sharadhr) [reply docs/diagrams/BasicRecipeDiagram.puml:83]: Maybe use `String`s for now, but keep a feature branch open that implements the `Unit` class properly

File: docs/diagrams/BasicRecipeDiagram.puml (hunk 1 above)
Comment (sharadhr) [reply docs/diagrams/BasicRecipeDiagram.puml:12]: Fixed.

**Pull request merged**